cat-l1b.sh

```diff
@@ -85,6 +93,6 @@ then
 fi
 
 #show contents
-$CONV -binfile $LOCALDIR/$DAT_PROD
+"$CONV" -binfile "$LOCALDIR/$DAT_PROD" [-ascfile "$LOCALDIR/$DAT_PROD"] 
```

Comment:
Is `[-ascfile "$LOCALDIR/$DAT_PROD"]` bash syntax? I read that the same as `test -ascfile "$LOCALDIR/$DAT_PROD"` (except that there should be blanks near [ and ]), and i don' understand the purpose.

Reply:
Sorry... the mistake should be [ -ascfile "$LOCALDIR/$DAT_PROD" ]

Reply:
Are you sure? `[ -ascfile "$LOCALDIR/$DAT_PROD" ]` sould be evaluated by the shell before passing the result to $CONV and it get:

```
 $ LOCALDIR=a; DAT_PROD=b; [ -ascfile "$LOCALDIR/$DAT_PROD" ]
-bash: [: -ascfile: unary operator expected
```

I don't think the shell knows what -ascfile is. 

I think the [ ] should not be there.

Reply:
Ah! I see. `[ -ascfile "$LOCALDIR/$DAT_PROD" ] ` is the usage of the $CONV. I tested w/ or w/o [  ] both work.



 ```
usage: software/RELEASE_2010-03-31/Bin2AsciiLevel1.e -binfile Level1_Bin_File [-ascfile Level1_Asc_File] [-nohead] [-nrec] [-head_only]

         [-masstnk tnk_id (3=both tanks,0=none)] [-massthr tnk_id (3=both tanks,0=none)]
         [-pres_tnk tnk_id] [-temp_tnk tnk_id]
         [-reg_tnk tnk_id] [-adap_tnk tnk_id] [-tempred_tnk tnk_id]
         [-tstart Tstart] [-tfinal Tfinal] [-acc1a_count] [-acc1a_ang] [-acc1a_lin] [-ahk1a_vp]
         [-acc1a_zbias Acc1aZbias (def 0)] [-ksnr] [-ka] [-k] [-rm_gdel] [-only_gdel]
         If -ascfile is not defined then output will go to standard out

```

Reply:
Ah ok, now I understand. Those `[ ]` (specifically in help screens like that one) usually mean that the enclosed argument is optional, it's merely an unspoken syntax that people have been using to explain what are optional input arguments. The same applies to all other input arguments of Bin2AsciiLevel1.e, they don't need to be specified.

Like it says there, if you don't pass the `-ascfile ... `arguments, the program spits it out to stdout. That's the point of `cat-l1b.sh`, just to spit out the L1B data. It's up to the users to redirect that output to whatever file they want

Reply:
Haha, thanks. My bad. I'll change it back.